### PR TITLE
set default extras as empty array to prevent compact() errors

### PR DESCRIPTION
**Pull request by kilip**
Hello,

PHP ```compact()``` function at line 413 will create php error when ```$args``` is an empty array:
https://github.com/zendframework/zend-view/blob/34babb457c8e38c64415b8615de39f9b3b5dcfbc/src/Helper/HeadLink.php#L379-L416

I have found this error in this travis build:
https://travis-ci.org/cross-solution/YAWIK/jobs/453775095#L1805
To reproducing error just simply execute this function ```HeadLink::prependStylesheet('/some/foo.css')```

I think this ```compact()``` error only occurs in php 7.3, but the code is buggy too. We need to define default value for ```$extras``` variables.

### CODE REVIEW

File: src/Helper/HeadLink.php [hunk 1]
```diff
@@ -408,6 +408,8 @@ public function createDataStylesheet(array $args)
         if ($args && is_array($args[0])) {
             $extras = array_shift($args);
             $extras = (array) $extras;
+        } else {
```
Comment (froschdesign) [inline src/Helper/HeadLink.php]: `else` is not needed here. Move the variable definition before `if` and use `$extras = []`.

File: src/Helper/HeadLink.php [hunk 2]
```diff
@@ -460,6 +462,8 @@ public function createDataAlternate(array $args)
             if (isset($extras['media']) && is_array($extras['media'])) {
                 $extras['media'] = implode(',', $extras['media']);
             }
+        } else {
```
Comment (froschdesign) [inline src/Helper/HeadLink.php]: Same here.

**Comment by froschdesign** [comment]
Reference for this change in PHP 7.3: http://php.net/manual/en/function.compact.php#refsect1-function.compact-changelog

**Comment by froschdesign** [comment]
Duplicate of #167

File: src/Helper/HeadLink.php [hunk 3]
```diff
@@ -405,6 +405,7 @@ public function createDataStylesheet(array $args)
             }
         }
 
+        $extras = [];
```
Comment (froschdesign) [inline src/Helper/HeadLink.php]: I'm sorry, but can you move this to line 386. Thanks!

File: src/Helper/HeadLink.php [hunk 4]
```diff
@@ -453,6 +454,7 @@ public function createDataAlternate(array $args)
         $type  = array_shift($args);
         $title = array_shift($args);
 
+        $extras = [];
```
Comment (froschdesign) [inline src/Helper/HeadLink.php]: And this line under the definition of `$title`.

File: src/Helper/HeadLink.php (hunk 3 above)
Comment (kilip) [reply src/Helper/HeadLink.php]: I have move the other $extra variable too to the default groups. please check

**Pull request closed without merging**